src/main/java/org/teacon/slides/render/ProjectorWorldRender.java

```diff
@@ -0,0 +1,94 @@
+import net.minecraftforge.client.event.RenderWorldLastEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import org.teacon.slides.ProjectorTileEntity;
+
+//Source: https://github.com/McJty/YouTubeModding14/blob/master/src/main/java/com/mcjty/mytutorial/client/InWorldRenderer.java
+@Mod.EventBusSubscriber(Dist.CLIENT)
+public class ProjectorWorldRender {
+    @SubscribeEvent
+    public static void worldRender(final RenderWorldLastEvent event) {
+        ClientPlayerEntity player = Minecraft.getInstance().player;
+        if (player.getHeldItemMainhand().getItem() == Items.STICK) {
+            locateProjectorTileEntities(player, event.getMatrixStack());
+        }
+    }
+
+    //To draw the line.
+    private static void drawLine(IVertexBuilder builder, Matrix4f positionMatrix, BlockPos pos, float dx1, float dy1, float dz1, float dx2, float dy2, float dz2, float red, float green, float blue, float alpha) {
+        builder.pos(positionMatrix, pos.getX() + dx1, pos.getY() + dy1, pos.getZ() + dz1)
+                .color(red, green, blue, alpha)
+                .endVertex();
+        builder.pos(positionMatrix, pos.getX() + dx2, pos.getY() + dy2, pos.getZ() + dz2)
+                .color(red, green, blue, alpha)
+                .endVertex();
+    }
+
+    private static void drawBlueLine(IVertexBuilder builder, Matrix4f positionMatrix, BlockPos pos, float dx1, float dy1, float dz1, float dx2, float dy2, float dz2) {
+        drawLine(builder, positionMatrix, pos, dx1, dy1, dz1, dx2, dy2, dz2, 1.0f, 1.0f, 1.0f, 1.0f);
+    }
+
+    private static void locateProjectorTileEntities(ClientPlayerEntity player, MatrixStack matrixStack) {
+        IRenderTypeBuffer.Impl buffer = Minecraft.getInstance().getRenderTypeBuffers().getBufferSource();
+        IVertexBuilder builder = buffer.getBuffer(ModRenderType.OVERLAY_LINES);
+
+        BlockPos playerPos = player.getPosition();
+        int px = playerPos.getX();
+        int py = playerPos.getY();
+        int pz = playerPos.getZ();
+        World world = player.getEntityWorld();
+
+        matrixStack.push();
+
+        Vec3d projectedView = Minecraft.getInstance().gameRenderer.getActiveRenderInfo().getProjectedView();
+        matrixStack.translate(-projectedView.x, -projectedView.y, -projectedView.z); // In default situation, the original position of matrix in world origin point. We should translate it.
+
+        Matrix4f matrix = matrixStack.getLast().getMatrix();
+
+        BlockPos.Mutable pos = new BlockPos.Mutable();
+        for (int dx = -10; dx <= 10; dx++) {
+            for (int dy = -10; dy <= 10; dy++) {
+                for (int dz = -10; dz <= 10; dz++) {
+                    pos.setPos(px + dx, py + dy, pz + dz);
+                    //Check ProjectorTileEntity
+                    if (world.getTileEntity(pos) != null && world.getTileEntity(pos).getType() == ProjectorTileEntity.theType) {
```

Comment:
I prefer to submit vertex data at TE Render stage rather than submit at RenderWorldLast stage ...

Besides, limiting the radius to 10 blocks sounds a bit unreasonable. All the projectors in the chunks to be rendered may need to be highlighted.

Reply:
I don't think submit vertex data at TE Render Stage is a good idea. We should render lines in top of buffer, it must be done at RenderWorldLast stage.

Reply:
Won't this cause render performance issues? I think @ustc-zzzz 's solution is more efficient.